detectable/src/main/java/com/synopsys/integration/detectable/detectables/git/GitExtractor.java

```diff
@@ -0,0 +1,74 @@
+package com.synopsys.integration.detectable.detectables.git;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.detectable.Extraction;
+import com.synopsys.integration.detectable.detectables.git.model.GitConfigElement;
+import com.synopsys.integration.detectable.detectables.git.parse.GitFileParser;
+import com.synopsys.integration.exception.IntegrationException;
+
+public class GitExtractor {
+    private final GitFileParser gitFileParser;
+
+    public GitExtractor(final GitFileParser gitFileParser) {
+        this.gitFileParser = gitFileParser;
+    }
+
+    public final Extraction extract(final File gitConfigFile, final File gitHeadFile) {
+        Extraction extraction;
+
```

Comment:
Worthy of a test?

Reply:
I moved most of the code to a GitFileTransformer which has a test.